prometheus/alert/client.go

```diff
@@ -61,23 +64,76 @@ func NewClient(fileLocks *FileLocker, prometheusURL string, fsClient fsclient.FS
 }
 
 // ValidateRule checks that a new alert rule is a valid specification
-func (c *client) ValidateRule(rule rulefmt.Rule) error {
+func ValidateRule(rule rulefmt.Rule) error {
 	// convert to RuleNode for validation
 	node := rulefmt.RuleNode{
 		Record:      yaml.Node{Value: rule.Record},
 		Alert:       yaml.Node{Value: rule.Alert},
 		Expr:        yaml.Node{Value: rule.Expr},
 		For:         0,
-		Labels:      nil,
-		Annotations: nil,
+		Labels:      rule.Labels,
+		Annotations: rule.Annotations,
 	}
-	errs := node.Validate()
-	if len(errs) != 0 {
-		return fmt.Errorf("invalid rule: %v", errs)
+	if len(node.Validate()) != 0 {
+		err := validateRuleImpl(node)
+		glog.Errorf("Invalid rule: %v", err)
+		return err
 	}
 	return nil
 }
 
+// validateRuleImpl determines the actual causes of the rule validation error.
+// Due to how the underlying prometheus types are made (unexported), we have to copy this code
```

Comment:
Can we update the prometheus types to be exported?

Reply:
No, they're owned by prometheus

Reply:
I know, but can we make a PR?